src/Pos/WorkMode.hs

```diff
@@ -194,3 +207,19 @@ instance
 instance SscHelpersClass ssc =>
          MonadBlockDBGenericWrite (BlockHeader ssc) (Block ssc) Undo (RealMode ssc) where
     dbPutBlund = dbPutBlundDefault
+
+instance MonadKnownPeers (RealMode ssc) where
+    updateKnownPeers f = do
+        oq <- rmcOutboundQ <$> ask
+        OQ.updateKnownPeers oq f
+    addKnownPeers peers = do
+        oq <- rmcOutboundQ <$> ask
+        OQ.updateKnownPeers oq (<> peers)
+    removeKnownPeer nid = do
+        oq <- rmcOutboundQ <$> ask
+        OQ.removeKnownPeer oq nid
+
+instance MonadFormatPeers (RealMode scc) where
+    formatKnownPeers formatter = do
+        oq <- rmcOutboundQ <$> ask
+        Just <$> OQ.dumpState oq formatter
```

Comment:
This logic is duplicated in few places. Can you define `updateKnownPeersDefault`, `addKnownPeersDefault`, `removeKnownPeerDefault`, `formatKnownPeersDefault` in one place and use them in all instances definitions?